### ISSUE TITLE
Bot Looping Forever (Doing Nothing).

### ISSUE THREAD

**Issue by becurrie** [becurrie/tt2_py#3]
Sometimes the bot will get stuck inside of the `in_transition_func()`. This could be improved slightly I'm sure.

```
[2019-01-31 08:44:40,688] INFO [bot.py:66 - __init__() ] bot instance has been initialized successfully
[2019-01-31 08:44:40,688] INFO [bot.py:67 - __init__() ] uuid: 0c29235e-33e9-43a5-97cb-665e6ceccbd4
[2019-01-31 08:44:40,689] INFO [bot.py:102 - _order_actions() ] actions in game will run in the following order
[2019-01-31 08:44:40,689] INFO [bot.py:104 - _order_actions() ] 1: level_heroes
[2019-01-31 08:44:40,689] INFO [bot.py:104 - _order_actions() ] 2: level_master
[2019-01-31 08:44:40,689] INFO [bot.py:104 - _order_actions() ] 3: level_skills
[2019-01-31 08:44:40,690] INFO [bot.py:119 - _order_skill_intervals() ] skills have been ordered successfully
[2019-01-31 08:44:40,690] INFO [bot.py:122 - _order_skill_intervals() ] 1: shadow_clone - 135
[2019-01-31 08:44:40,690] INFO [bot.py:122 - _order_skill_intervals() ] 2: war_cry - 130
[2019-01-31 08:44:40,690] INFO [bot.py:122 - _order_skill_intervals() ] 3: hand_of_midas - 60
[2019-01-31 08:44:40,690] INFO [bot.py:122 - _order_skill_intervals() ] 4: heavenly_strike - 0
[2019-01-31 08:44:40,691] INFO [bot.py:122 - _order_skill_intervals() ] 5: deadly_strike - 0
[2019-01-31 08:44:40,691] INFO [bot.py:122 - _order_skill_intervals() ] 6: fire_sword - 0
[2019-01-31 08:44:40,691] INFO [bot.py:145 - calculate_skill_execution() ] attempting to calculate next skill execution times
[2019-01-31 08:44:40,691] INFO [bot.py:155 - calculate_skill_execution() ] next hand_of_midas will be activated at: 2019-01-31 08:45:40.691356 (0d 00:01:00)
[2019-01-31 08:44:40,692] INFO [bot.py:155 - calculate_skill_execution() ] next war_cry will be activated at: 2019-01-31 08:46:50.691356 (0d 00:02:10)
[2019-01-31 08:44:40,692] INFO [bot.py:155 - calculate_skill_execution() ] next shadow_clone will be activated at: 2019-01-31 08:46:55.691356 (0d 00:02:15)
[2019-01-31 08:44:40,692] INFO [bot.py:166 - calculate_next_prestige() ] next prestige will take place at: 2019-01-31 09:54:40.692332 (0d 01:10:00)
[2019-01-31 08:44:40,692] INFO [bot.py:184 - calculate_next_stats_update() ] next stats update run will take place at: 2019-01-31 09:14:40.692332 (0d 00:30:00)
[2019-01-31 08:44:45,425] INFO [bot.py:712 - run() ] running bot now...
```

**Comment by becurrie** [comment]
`in_transition_func` should be re written to check for a state of transition, and fix it appropriately.

- Ad on screen.
- Stage changing.
- Window opened with "X" available on top half of screen.
- Clan icon on screen.

Any of these possibilities would represent either a state to leave from, or a state where we should click somewhere to get out of that screen.

**Comment by becurrie** [comment]
Likely resolved by #24
